temporalcli/commands.activity.go

```diff
@@ -189,3 +189,62 @@ func (c *TemporalActivityUpdateOptionsCommand) run(cctx *CommandContext, args []
 
 	return nil
 }
+
+func (c *TemporalActivityPauseCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	_, err = cl.WorkflowService().PauseActivityById(cctx, &workflowservice.PauseActivityByIdRequest{
+		Namespace:  c.Parent.Namespace,
+		WorkflowId: c.WorkflowId,
+		RunId:      c.RunId,
+		ActivityId: c.ActivityId,
+
+		Identity: c.Identity,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to update Activity options: %w", err)
+	}
+
+	return nil
+}
+
+func (c *TemporalActivityUnpauseCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	request := &workflowservice.UnpauseActivityByIdRequest{
+		Namespace:  c.Parent.Namespace,
+		WorkflowId: c.WorkflowId,
+		RunId:      c.RunId,
+		ActivityId: c.ActivityId,
+		Identity:   c.Identity,
+	}
+	if c.Reset {
```

Comment:
For me to understand not related to CLI specifically, what would happen server-side if I didn't set either of the `oneof`s for operation? I couldn't tell from proto docs.

Reply:
Server side:
it neither is provided - activity will be "unpaused', but will not start if it in retry.

Reply:
To confirm, that is the equivalent of providing an empty resume in the oneof? I just want to make sure there isn't a state not represented here?

Reply:
Not sure what you mean by "empty resume" - the one that doesn't have noWait set?
If so - yes, resume without noWait set and no flags at all will lead to the same result - activity will be unpaused, but will not be forcibly immediately started (it still can if it is past retry time).

Reply:
> Not sure what you mean by "empty resume" - the one that doesn't have noWait set?

Yeah, a proto object with all of its default fields (so yes no_wait unset which defaults to false). So the absent oneof and a no-wait-false resume are the same thing in the API. If you do make any new commits to the API repo, we should add a comment clarifying that the default is resume.